semestre-service/src/main/java/ma/enset/semestreservice/service/SemestreServiceImpl.java

```diff
@@ -97,6 +97,13 @@ public void deleteAllByCodeSemestre(List<String> codesSemestres) throws ElementN
         codesSemestres.forEach(this::deleteByCodeSemestre);
     }
 
+    @Override
+    public List<Semestre> findAllByCodesOfSemestres(List<String> codesSemestres) throws ElementNotFoundException {
+        List<Semestre> semestres = new ArrayList<>();
+        codesSemestres.forEach(code -> semestres.add(findByCodeSemestre(code)));
```

Comment:
We decided to fetch all the records using the repository method `findAll` and then check the existsance of the ids within the retrieved set as the way we were handling it is not efficient.
Currently some services have fiixed this issue and it will be fixed for the others in the future.

Reply:
When you retrieve all records and filter them in the application code, you are transferring more data than necessary between the database and the application. This can lead to increased network traffic and slower response times, especially if the dataset is large.

Reply:
When you retrieve all records and filter them in the application code, you are transferring more data than necessary between the database and the application. This can lead to increased network traffic and slower response times, especially if the dataset is large , so i will not change it

Reply:
The first part of your response is true if it's based on the fact that I had a typo in my comment where I wrote `findAll` instead of `findAllById`. I don't believe it's still the case after the fix.

> so i will not change it

Do not feel pressured, I'll implement it once I finish the tasks I'm currently working on.